rpcserver.go

```diff
@@ -5166,13 +5166,16 @@ func marshalDbEdge(edgeInfo *channeldb.ChannelEdgeInfo,
 
 	// Make sure the policies match the node they belong to. c1 should point
 	// to the policy for NodeKey1, and c2 for NodeKey2.
-	if c1.ChannelFlags&lnwire.ChanUpdateDirection == 1 {
+	flipped := false
+	if c1 != nil && c1.ChannelFlags&lnwire.ChanUpdateDirection == 1 {
 		c2, c1 = c1, c2
+		flipped = true
 	}
 
-	// Order the edges by increasing pubkey.
-	if bytes.Compare(edgeInfo.NodeKey2Bytes[:],
-		edgeInfo.NodeKey1Bytes[:]) < 0 {
+	// In case c1 was nil we look at c2. But if we already flipped the
+	// direction, let's not do it again.
+	if !flipped && c2 != nil &&
+		c2.ChannelFlags&lnwire.ChanUpdateDirection == 1 {
```

Comment:
We want to check `c2.ChannelFlags&lnwire.ChanUpdateDirection == 0` instead, as that would point to the second policy actually belonging to the first node. We can also remove the flipped logic as it's not necessary.

Reply:
Ah, makes sense. Updated.